src/FileConventions/Library.fs

```diff
@@ -329,3 +329,46 @@ let DetectInconsistentVersionsInFSharpScripts
         false
     else
         DetectInconsistentVersionsInNugetRefsInFSharpScripts fsxFiles
+
+let allowedNonVerboseFlags =
+    seq {
+        "unzip"
+
+        // even if env in linux has --split-string=foo as equivalent to -S, it
+        // doesn't seem to be present in macOS' env man page and doesn't work either
+        "env -S"
+    }
+
+let NonVerboseFlagsInGitHubCI(fileInfo: FileInfo) =
+    let validExtensions =
+        seq {
+            ".yml"
+            ".fsx"
+            ".fs"
+            ".sh"
+        }
+
+    assert
+        validExtensions
+        |> Seq.map(fun ext -> fileInfo.FullName.EndsWith(ext))
+        |> Seq.contains true
```

Comment:
@realmarv can you explain what exactly are you doing here please? this is a bit unreadable; also, why are you using assert instead of failwith? I'm actually not familiar with `assert` func TBH, but I have the feeling that it only works in DEBUG mode and the error thrown would be very unreadable (as opposed to using failwith where you are required to provide a readable error)

Reply:
@realmarv BTW I just cooked this in case you want to use it: https://github.com/nblockchain/fsx/commit/7acb7a5b8fa374ee2b68c28c2e438cb6c9265e58 (will be available in nuget in 5min)

Reply:
It loops over validExtensions sequence and checks if the file path ends with any of them